### PR TITLE
Add network identifier to cryptography - Closes #4333

### LINKED ISSUE #4333: Add getNetworkIdentifier function
### Description

[LIP-0009](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0009.md) adds a function to calculate network identifier, and it should be added to cryptography library

- `network identifier = H(nethash + community identifier)`

### Motivation

Motivation is described in [LIP-0009](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0009.md)

### Acceptance Criteria

- Reviewed from research team
- Tested all possibility

### PR DESCRIPTION
**Pull request by mitsujutsu**
### What was the problem?

`network identifier = H(nethash + community identifier)`

### How did I solve it?

Added `getNetworkIdentifier`

### How to manually test it?

`npm run test` in `lisk-cryptography`

### Review checklist

- [ ] The PR resolves #4333 
- [ ] All new code is covered with unit tests
- [ ] Relevant integration / functional tests are added
- [ ] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [ ] Documentation has been added/updated

### CODE REVIEW

File: elements/lisk-cryptography/src/hash.ts [hunk 1]
```diff
@@ -43,3 +43,8 @@ export const hash = (data: Buffer | string, format?: string): Buffer => {
 		'Unsupported data format. Currently only Buffers or `hex` and `utf8` strings are supported.',
 	);
 };
+
+export const getNetworkIdentifier = (
```
Comment (pablitovicente) [inline elements/lisk-cryptography/src/hash.ts]: Could you add tests for checking behaviour when the inputs are `null`, `undefined` and `''` (empty string) and maybe consider adding validation to not allow any of those values? Of course in TS it's not a problem but when using the function from JS we can pass anything. 

Otherwise looks good to go!

Comment (mitsujutsu) [reply elements/lisk-cryptography/src/hash.ts]: These types are already checked for in the `hash` function. The hash function will throw if the input is anything other than `string` or `Buffer`.
